manifests/unit_file.pp

```diff
@@ -121,6 +125,7 @@
       provider   => 'systemd',
       hasrestart => $hasrestart,
       hasstatus  => $hasstatus,
```

Comment:
I do think we should deprecate these parameters in favor of `$service_parameters`.

Reply:
I considered it, but that would be a breaking change. I propose we merge this PR and I create a new one that's a breaking change and we merge that in the future?

Reply:
we cold also already implement something like `warn('hastatus is deprecated')`?

Reply:
I said deprecate, not remove :)

So what you can do is to add a note in the `@param` tag. Then in the code:
```puppet
$deprecated_service_params = {
  'hasrestart' => $hasrestart,
  'hasstatus'  => $hasstatus,
}

service { '...':
  * => $deprecated_service_params + $service_params,
}
```

Reply:
I implemented this pattern already. For a clearer changelog I would like to add the deprecation in another PR.